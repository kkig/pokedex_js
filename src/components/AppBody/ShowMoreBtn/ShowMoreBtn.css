.more-btn-section {
    display: flex; 
    justify-content: center;
    padding: 1em 0;
}

.more-btn:hover {
    color: white;
}