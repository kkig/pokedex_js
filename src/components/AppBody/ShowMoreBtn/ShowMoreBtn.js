import React from 'react';

import './ShowMoreBtn.css';

// Material UI
import Button from '@material-ui/core/Button';

const ShowMoreBtn = ({ endpoint, handleClick }) => {

    return (

        // Only display button when there is more list to show

        <div className="more-btn-section" data-testid="btn-container">
            {
                endpoint !== null ? 
            <Button 
                variant="contained" 
                className="more-btn"
                data-testid="show-more-btn"
                onClick={handleClick}
            >
                Show More
            </Button>   :

                null
            }
        </div>

    );
};

export default ShowMoreBtn;