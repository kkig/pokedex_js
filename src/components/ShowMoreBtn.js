import React from 'react';

import './ShowMoreBtn.css';

// Material UI
import Button from '@material-ui/core/Button';

const ShowMoreBtn = props => {
    return (
        <div className="more-btn-section">
            <Button 
                variant="contained" 
                color="primary" 
                className="more-btn"
                onClick={props.handleClick}
            >
                Show More
            </Button>
        </div>
    );
};

export default ShowMoreBtn;