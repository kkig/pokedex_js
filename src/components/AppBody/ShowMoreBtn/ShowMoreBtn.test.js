import React from 'react';
import { render, cleanup } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';

import ShowMoreBtn from './ShowMoreBtn';

afterEach(cleanup);

it("renders", () => {
    const { asFragment } = render(<ShowMoreBtn endpoint="https://pokeapi.co/api/v2/pokemon/?limit=20" />);
    expect(asFragment()).toMatchSnapshot();
});


it("when there is more list, button will be displayed", () => {
    const { getByTestId } = render(<ShowMoreBtn endpoint="https://pokeapi.co/api/v2/pokemon/?limit=20" />);

    expect(getByTestId('show-more-btn')).toBeVisible();
});


it("when there is no more list to add, button will not be visible", () => {
    const { getByTestId } = render(<ShowMoreBtn endpoint={null} />);
    expect(getByTestId('btn-container')).toBeEmpty();
});



