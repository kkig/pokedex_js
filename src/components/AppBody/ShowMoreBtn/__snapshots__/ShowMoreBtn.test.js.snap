// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
<DocumentFragment>
  <div
    class="more-btn-section"
    data-testid="btn-container"
  >
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-contained more-btn MuiButton-containedPrimary"
      data-testid="show-more-btn"
      tabindex="0"
      type="button"
    >
      <span
        class="MuiButton-label"
      >
        Show More
      </span>
      <span
        class="MuiTouchRipple-root"
      />
    </button>
  </div>
</DocumentFragment>
`;
